src/languages/perl.js

```diff
@@ -258,19 +261,44 @@ export default function(hljs) {
     end: /\}/
     // contains defined later
   };
-  const VAR = { variants: [
-    { begin: /\$\d/ },
-    { begin: regex.concat(
-      /[$%@](\^\w\b|#\w+(::\w+)*|\{\w+\}|\w+(::\w*)*)/,
-      // negative look-ahead tries to avoid matching patterns that are not
-      // Perl at all like $ident$, @ident@, etc.
-      `(?![A-Za-z])(?![@$%])`
-    ) },
-    {
-      begin: /[$%@][^\s\w{]/,
-      relevance: 0
-    }
-  ] };
+  const ATTR = {
+    className: 'attribute',
+    begin: /\s+:\s*\w+/,
+    end: /(\\s*\\(.*?\\))?/,
+  };
+  const VAR = {
+    variants: [
+      { begin: /\$\d/ },
+      { begin: regex.concat(
+        /[$%@](\^\w\b|#\w+(::\w+)*|\{\w+\}|\w+(::\w*)*)/,
+        // negative look-ahead tries to avoid matching patterns that are not
+        // Perl at all like $ident$, @ident@, etc.
+        `(?![A-Za-z])(?![@$%])`
+        )
+      },
+      {
+        begin: /[$%@][^\s\w{]/,
+        relevance: 0
+      }
+    ],
+    contains: [ ATTR ],
+  };
+  const NUMBER = {
+    className: 'number',
+    variants: [
+      // decimal numbers:
+      // include the case where a number starts with a dot (eg. .9), and
+      // the leading 0? avoids mixing the first and second match on 0.x cases
+      { match: /0?\.[0-9][0-9_]+\b/ },
+      // include the special versioned number (eg. v5.38)
+      { match: /\bv?(0|[1-9][0-9_]*(\.[0-9_]+)?|[1-9][0-9_]*)\b/ },
```

Comment:
Does the `v` variant only work after `use`?

Reply:
Well observed. Yes, that's specific to `use`. However, thinking a bit more on that, it would be better to remove any matching with number, as it's not actually a number, but a package name. I want to remove the current behavior of highlighting half the number (`.38` is highlighted) and have no highlighting at all:
```perl
use v5.38;
```
But I'm not sure how to do it.

Reply:
At first, I thought so, but later I realized classes can also be versioned with the `v` in front of the number, like `class MyClass v1.0.0 {..}` and, consequently, these versioned classes can be inherited with `:isa(MyClass v1.0.0)` (`:isa` is an attribute).

Reply:
We can leave it and see how it works.